CMSgov/onemac
# Update Details Cards UI

Story: https://qmacbis.atlassian.net/browse/OY2-22082

### Details

This PR:
- Extracts the `<DetailCard>` component for re-use
- Updates the UI on the Details page to show two `<DetailCard>`s (previously one unified card)

### Changes

- New file `DetailCard.tsx` contains extracted card component. This component is meant to be fed children and apply the card-like styling (i.e. the gradient-topped box) to whatever it is given.
- Updates to CSS for layout and scaling of cards
- Updates to JSX for layout

| Version | VAL | OY2-22082 |
| ----- | ----- | ----- |
| Desktop | <img width="250" alt="Screen Shot 2023-04-03 at 12 31 53 PM" src="https://user-images.githubusercontent.com/4022304/229571635-0e9aa495-4007-43d8-99a3-da5fa60ddc1c.png"> | <img width="250" alt="Screen Shot 2023-04-03 at 12 21 09 PM" src="https://user-images.githubusercontent.com/4022304/229571184-9ace1b36-b626-4703-b0f1-80e0d76a5729.png"> |
| Tablet | <img width="250" alt="Screen Shot 2023-04-03 at 12 31 44 PM" src="https://user-images.githubusercontent.com/4022304/229571665-93b0b2de-a411-4a4b-9139-241372d34316.png"> | <img width="250" alt="Screen Shot 2023-04-03 at 12 21 28 PM" src="https://user-images.githubusercontent.com/4022304/229571270-7e42132e-3df8-44f3-ad91-d49ea001db13.png"> |
| Mobile | <img width="250" alt="Screen Shot 2023-04-03 at 12 31 37 PM" src="https://user-images.githubusercontent.com/4022304/229571728-471358cf-2018-4c83-b634-12ea6fd1a044.png"> | <img width="250" alt="Screen Shot 2023-04-03 at 12 21 36 PM" src="https://user-images.githubusercontent.com/4022304/229571316-decf793e-b6f2-4eb6-b922-4099ebb7e2c4.png"> |

### Implementation Notes

- When possible, I'll be breaking out components like this to cut back on how much JSX we hand-write

### Test Plan

1. Pull latest and run `npm install` (for good measure)
2. Arrive at the details page of a SPA and/or Waiver by logging in as a test user and navigating to the dashboard, then clicking the hyperlink ID
3. Observe the new UI and ensure it matches the designs given
4. Use browser dev tools (right-click > inspect, or F12) to scale the size to view scaling styles on all screen sizes. Ensure no styling breaks
